src/pages/recipes/AddIngredient.jsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import * as yup from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { setLocale } from "yup";
+import { useNavigate } from "react-router-dom";
+import propTypes from "prop-types";
+import { Input } from "../../components/Input/Input";
+import { Container } from "../../components/Container/Container";
+import { Button } from "../../components/Button/Button";
+import { Text } from "../../components/Text/Text";
+import { ButtonGroup } from "../../components/ButtonGroup/ButtonGroup";
+
+function AddIngredient({ onSubmit }) {
+  setLocale({
+    mixed: {
+      required: "Pole ${label} jest wymagane.",
+    },
+    number: {
+      moreThan: "Pole ${label} musi mieć wartość większą od ${more}.",
+    },
+  });
+
+  const schema = yup.object({
+    name: yup.string().required().label("nazwa"),
+    amount: yup
+      .number()
+      .transform((value) => (Number.isNaN(value) ? 0 : value))
+      .moreThan(0)
+      .label("ilość"),
+    measurment: yup.string().required().label("miara"),
+  });
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ resolver: yupResolver(schema) });
+
+  const navigate = useNavigate();
+
+  const handleBackButtonClick = () => {
+    navigate("/recipe/new");
```

Comment:
maybe we could use `Link` instead of `navigate`?

Reply:
I wanted to use `navigate` to avoid anchor default styling.